pysmt/solvers/btor.py

```diff
@@ -175,7 +175,10 @@ def __init__(self, environment, logic, **options):
 
     @clear_pending_pop
     def _reset_assertions(self):
-        raise NotImplementedError
+        self.btor = pyboolector.Boolector()
```

Comment:
Can we do better here? Isn't there a way to avoid reconstructing the solver?
For example, Mathsat has a msat_reset_env method, and z3 has a z3_reset method. If not, this is ok.

Reply:
I'm not sure actually -- I don't see an obvious way at least. I'll look into it and get back to you!

Reply:
Looks like there isn't a different way currently, but it's an open issue for them: https://github.com/Boolector/boolector/issues/32.

Reply:
Ok, no action required, then.